src/Sarif.Viewer.VisualStudio/ErrorList/SarifResultTableEntry.cs

```diff
@@ -0,0 +1,225 @@
+﻿namespace Microsoft.Sarif.Viewer.ErrorList
+{
+    using EnvDTE;
+    using Microsoft.CodeAnalysis.Sarif;
+    using Microsoft.Sarif.Viewer.Models;
+    using Microsoft.VisualStudio.Shell;
+    using Microsoft.VisualStudio.Shell.Interop;
+    using Microsoft.VisualStudio.Shell.TableControl;
+    using Microsoft.VisualStudio.Shell.TableManager;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Windows;
+    using System.Windows.Documents;
+
+    internal sealed class SarifResultTableEntry : ITableEntry
+    {
+        private readonly Dictionary<string, object> columnKeyToContent = new Dictionary<string, object>(StringComparer.InvariantCulture);
+
+        public static readonly string[] SupportedColumns = new[]
```

Comment:
>readonly [](start = 22, length = 8)

Can this be const?

Reply:
But the elements could still be changed. Apply `AsReadOnly` to produce a `ReadOnlyCollection`.

---
In reply to: [473244487](https://github.com/microsoft/sarif-visualstudio-extension/pull/172#discussion_r473244487) [](ancestors = 473244487)